### PR TITLE
Hero slider

### PR DESCRIPTION
**Pull request by koukouchris**

### CODE REVIEW

File: Components/HeroSlider/functions.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                                'type' => 'image',
+                                'return_format' => 'array',
+                                'preview_size' => 'medium',
+                                'library' => 'all',
+                                'mime_types' => 'jpg,jpeg',
+                                'required' => 1,
+                                'instructions' => 'Recommended resolution greater than 2048 x 800 px.'
+                            ],
+                            [
+                                'label' => 'Desktop Align Image',
```
Comment (steffenbew) [inline Components/HeroSlider/functions.php]: 'Image Alignment'?

File: Components/HeroSlider/functions.php [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                                'name' => 'imageMobile',
+                                'type' => 'image',
+                                'return_format' => 'array',
+                                'preview_size' => 'medium',
+                                'library' => 'all',
+                                'mime_types' => 'jpg,jpeg',
+                                'instructions' => 'Recommended resolution greater than 750 x 800 px.'
+                            ],
+                            [
+                                'label' => 'Mobile Align Image',
```
Comment (steffenbew) [inline Components/HeroSlider/functions.php]: 'Image Alignment'?

File: Components/HeroSlider/index.twig [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<div class="flyntComponent themeDark" is="flynt-hero-slider">
+  <script type="application/json">{{ jsonData|json_encode }}</script>
+  <div data-slider class="slider swiper-container">
+    <div class="swiper-wrapper">
+      {% for slide in slides %}
+      <div class="swiper-slide">
+        <picture class="content-picture">
```
Comment (steffenbew) [inline Components/HeroSlider/index.twig]: Class hierarchy seems to be a little weird to me, as `.content` is on the same dom level down below. The naming would imply to me that this is a child of `.content`.

File: Components/HeroSlider/index.twig [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          <source srcset="
+            {{ slide.images.imageDesktop.src | resize(2560, 800, 'center') }} 2560w,
```
Comment (steffenbew) [inline Components/HeroSlider/index.twig]: I believe we should use `dynamicResize` here (and everywhere).

File: Components/HeroSlider/script.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  connectedCallback () {
+    if (this.$sliderItems.length > 1) {
+      this.initSlider()
+    }
+  }
+
+  initSlider () {
+    const { options } = this.props
+    const config = {
+      a11y: options.a11y,
```
Comment (steffenbew) [inline Components/HeroSlider/script.js]: Is this property actually set?

Comment (emcarru) [reply Components/HeroSlider/script.js]: Yes, it's pulled through from Translatable Options -> SliderOptions

File: Components/HeroSlider/README.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+Category: Slider
+
+---
+
+Tags: image, photo, gallery, content
+
+---
+
+Text: Full width image slider with navigation arrows. Upload two versions of image, one for desktop one for mobile and optionaly add html content overflowing the image. This component uses swiper slider.
```
[the remark this answers is not in the document]
Comment (emcarru) [reply Components/HeroSlider/README.md]: updated

File: Components/HeroSlider/functions.php [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+                            [
+                                'label' => 'Desktop Align Image',
+                                'name' => 'desktopAlignImage',
+                                'type' => 'select',
+                                'allow_null' => 0,
+                                'multiple' => 0,
+                                'ui' => 1,
+                                'ajax' => 0,
+                                'choices' => [
+                                    'content-image--isCentered' => 'Center',
```
[the remark this answers is not in the document]
Comment (emcarru) [reply Components/HeroSlider/functions.php]: updated without 'is' and also fixed consistency (turns out, center on mobile wasn't actually doing anything)

File: Components/HeroSlider/functions.php (hunk 2 above)
Comment (emcarru) [reply Components/HeroSlider/functions.php]: updated

File: Components/HeroSlider/functions.php [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+                            [
+                                'label' => 'Mobile Align Image',
+                                'name' => 'mobileAlignImage',
+                                'type' => 'select',
+                                'allow_null' => 0,
+                                'multiple' => 0,
+                                'ui' => 1,
+                                'ajax' => 0,
+                                'choices' => [
+                                    'content-image--isCentered' => 'Center',
```
[the remark this answers is not in the document]
Comment (emcarru) [reply Components/HeroSlider/functions.php]: updated

File: Components/HeroSlider/index.twig (hunk 3 above)
Comment (emcarru) [reply Components/HeroSlider/index.twig]: Yeah, moved it into its own context now

File: Components/HeroSlider/index.twig (hunk 4 above)
Comment (emcarru) [reply Components/HeroSlider/index.twig]: done

**Pull request merged**